src/lib/apis/fetch.ts

```diff
@@ -71,8 +71,18 @@ export default (url, options = {}) => {
     const { method } = opts
     const { url: cleanedUrl, body, json } = constructUrlAndParams(method, url)
 
+    const isBrazeRequest = userAgent.includes("Braze")
+
     request(cleanedUrl, { ...opts, body, json }, (err, response) => {
       if (err) return reject(err)
+
+      // Any Braze requests that resulted in a 429 should propagate that status
+      // all the way up (entire MP response should be a 429).
+      // That way Braze knows to retry.
+      if (isBrazeRequest && response.statusCode === 429) {
+        return reject(new HTTPError("Braze rate limit reached", 429))
```

Comment:
The API intentionally isn't vendor-specific (calling this "email provider"), and in this intermediate layer we drop that abstraction and call it `isBrazeRequest` and `Braze rate limit reached`. Could we instead use the same language all the way through? It's minor (and arguably a little _less_ obvious to clients) but more consistent and I think ultimately clear.

Reply:
I actually wanted to use the 'Braze' lingo here - the 'backend' (Gravity) can keep the `email_provider` abstraction, but MP (still a backend, but less so - it's downstream from Gravity and is a _client_ of Gravity's) would use this more explicit form.

Sort-of arbitrary no right/wrong answer!

I'm actually taking a detour to try an alternate implementation from this PR, but there'd need to be specific naming there as well to propagate this scenario properly - I can use 'email provider' in that version.